students/briggsm/session04/mb_trigram.py

```diff
@@ -0,0 +1,54 @@
+import random as rand
+
+def create_corpus(infile):
+    '''Input: a filename
+       Output: a list of words.'''
+    corpus = []
+    with open(infile) as lines:
+        lines.readline()
+        for line in lines:
+            line = line.strip()
+            words = line.split(" ")
+            for i in range(len(words)):
+                corpus.append(words[i])
+    return corpus
+
+def create_trigram(textlist):
+    '''Input: a list of words from a text body.
+       Output: a dictionary of two words in sequence and the word after.'''
+    trigram = {}
+    for w in range(len(textlist[:-2])):
+        firstword = textlist[w]
+        secondword = textlist[w+1]
+        thirdword = textlist[w+2]
+        trigram[thirdword] = (firstword, secondword)
```

Comment:
I think this is backwards -- you have the third word as the key and the first two as the value, so now you can look up a word and see what two words preceded it. Which is not what you want.

Plus, every time you hit the same word again, it will overwrite the last occurrence!

you (almost) want, in the end, something like:

trigram[(firstword, secondword)] = [a_previous_word, another_one, third_word]

that is -- the keys are the first two words, and the value is a list of all the words that have followed that pair of words.